x-pack/plugins/infra/public/components/asset_details/hooks/use_metadata.ts

```diff
@@ -51,10 +55,12 @@ export function useMetadata({
   );
 
   useEffect(() => {
-    (async () => {
-      await makeRequest();
-    })();
-  }, [makeRequest]);
+    if (request$) {
+      request$.next(makeRequest);
+    } else {
+      makeRequest();
```

Comment:
Why not just use the `loading` flag to prevent the auto-refresh from canceling ongoing requests? 

When the auto-refresh is turned off, if users click on the submit button multiple times in short intervals, the asset details component has to make new requests. If there are pending ongoing requests, these must be aborted.

With auto-refresh-turned on we must prevent requests from being aborted, but we also need to control when new requests can be made. That's why I'm delegating that to the `request$` observer